### PR TITLE
Enable and fix a number of standard lints and deprecated member use

### PR DESCRIPTION
**Pull request by kevmoo**
Improves the package health when analyzed on the pub site

### CODE REVIEW

**Comment by kevmoo** [comment]
@munificent trying to cleanup https://pub.dartlang.org/api/packages/args/metrics?reports

File: lib/src/arg_parser.dart [hunk 1]
```diff
@@ -2,6 +2,10 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+// TODO(nweiz): Remove this ignore when sdk#30084 is fixed or when args no
```
Comment (munificent) [inline lib/src/arg_parser.dart]: "kevmoo", unless you're unilaterally signing up Natalie for work. :)

File: test/args_test.dart [hunk 2]
```diff
@@ -112,15 +113,20 @@ void main() {
     test(
         'throws ArgumentError if splitCommas is passed with allowMultiple: '
         'false', () {
-      var parser = new ArgParser();
-      throwsIllegalArg(() => parser.addOption('flummox', splitCommas: true));
-      throwsIllegalArg(() => parser.addOption('flummox', splitCommas: false));
+      var parser = ArgParser();
+      throwsIllegalArg(() => parser.addOption(
```
Comment (munificent) [inline test/args_test.dart]: Now that there's a line comment in there, I think this is clearer as a `{ ... }` lambda. Below too.

File: lib/src/arg_parser.dart (hunk 1 above)
Comment (kevmoo) [reply lib/src/arg_parser.dart]: This was a copy-paste from analysis_options, so I left the owner as-is.

File: test/args_test.dart (hunk 2 above)
Comment (kevmoo) [reply test/args_test.dart]: ack

**Pull request merged**